plugins/module_utils/network/sonic/config/ip_neighbor_interfaces/ip_neighbor_interfaces.py

```diff
@@ -0,0 +1,369 @@
+        else:
+            for conf in want:
+                intf_name = conf['name']
+                have_conf = search_obj_in_list(intf_name, have, 'name')
+                if not have_conf:
+                    continue
+
+                # Delete all neighbor config if only interface name is specified
+                if len(conf.keys()) == 1:
+                    commands.append(conf)
+                    requests.extend(self.get_delete_ip_neighbor_interfaces_requests(conf, have_conf, True))
+                    continue
+
+                command = {}
+                for option in ('ipv4_neighbors', 'ipv6_neighbors'):
+                    if conf.get(option) and have_conf.get(option):
+                        conf_neighbors = {neighbor['ip'] for neighbor in conf[option]}
+                        have_neighbors = {neighbor['ip'] for neighbor in have_conf[option]}
+                        del_neighbors = conf_neighbors.intersection(have_neighbors)
+                        if del_neighbors:
+                            command[option] = [{'ip': neighbor} for neighbor in del_neighbors]
+
+                if command:
+                    command['name'] = conf['name']
+                    commands.append(command)
+                    requests.extend(self.get_delete_ip_neighbor_interfaces_requests(command, have_conf))
+
+        if commands:
+            commands = update_states(commands, 'deleted')
+        return commands, requests
+
+    def get_delete_ip_neighbor_interfaces_requests(self, command, have_conf, delete_all=False):
+        requests = []
+        if delete_all or len(command.keys()) == 1:
+            for option in ('ipv4_neighbors', 'ipv6_neighbors'):
+                if have_conf.get(option):
+                    requests.append({'path': self.get_neighbors_path(command['name'], option), 'method': DELETE})
+        else:
+            for option in ('ipv4_neighbors', 'ipv6_neighbors'):
+                if option not in command:
+                    continue
+
+                url = self.get_neighbors_path(command['name'], option)
+                for neighbor in command[option]:
+                    requests.append({'path': url + '/neighbor=' + neighbor['ip'], 'method': DELETE})
+
+        return requests
+
+    def get_modify_ip_neighbor_interfaces_requests(self, commands):
+        requests = []
+
+        for command in commands:
+            name = command['name']
+            for option in ('ipv4_neighbors', 'ipv6_neighbors'):
+                if option not in command:
+                    continue
+
+                url = self.get_neighbors_path(command['name'], option)
+                for neighbor in command[option]:
+                    payload = {
+                        'openconfig-if-ip:neighbors': {
+                            'neighbor': [{
+                                'ip': neighbor['ip'],
+                                'config': {'ip': neighbor['ip'], 'link-layer-address': neighbor['mac']}
```

Comment:
'mac' is optional. If isn't specified in the playbook, won't this cause an exception?

Reply:
Documentation and input validation has been added to ensure both 'ip' and 'mac' are present for states other than deleted.